airbyte-integrations/integration-test-lib/src/main/java/io/airbyte/integrations/base/TestDestination.java

```diff
@@ -160,35 +157,73 @@ void testCheckConnectionInvalidCredentials() {
     // todo (cgardens) - blocked on worker not calling discover worker.
   }
 
+  private static class DataArgumentsProvider implements ArgumentsProvider {
+
+    @Override
+    public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+      return Stream.of(
+          Arguments.of("exchange_rate_messages.txt", "exchange_rate_catalog.json")
+      // todo - need to use the new protocol to capture this.
+      // Arguments.of("stripe_messages.txt", "stripe_schema.json")
+      );
+    }
+
+  }
+
   /**
    * Verify that the integration successfully writes records. Tests a wide variety of messages and
    * schemas (aspirationally, anyway).
    */
+  @ParameterizedTest
+  @ArgumentsSource(DataArgumentsProvider.class)
+  void testSync(String messagesFilename, String catalogFilename) throws Exception {
+    final Schema catalog = Jsons.deserialize(MoreResources.readResource(catalogFilename), Schema.class);
+    final List<SingerMessage> messages = MoreResources.readResource(messagesFilename).lines()
+        .map(record -> Jsons.deserialize(record, SingerMessage.class)).collect(Collectors.toList());
+    runSync(messages, catalog);
+
+    assertSameMessages(messages, recordRetriever(testEnv));
+  }
+
+  /**
+   * Verify that the integration overwrites the first sync with the second sync.
+   */
   @Test
-  void testSync() throws Exception {
-    final DefaultSingerTarget target = new DefaultSingerTarget(getImageName(), pbf);
-    final List<SingerMessage> messages = MoreResources.readResource("messages.txt").lines()
+  void testSecondSync() throws Exception {
+    final Schema catalog = Jsons.deserialize(MoreResources.readResource("exchange_rate_catalog.json"), Schema.class);
+    final List<SingerMessage> firstSyncMessages = MoreResources.readResource("exchange_rate_messages.txt").lines()
         .map(record -> Jsons.deserialize(record, SingerMessage.class)).collect(Collectors.toList());
+    runSync(firstSyncMessages, catalog);
+
+    List<SingerMessage> secondSyncMessages = Lists.newArrayList(new SingerMessage()
+        .withStream(catalog.getStreams().get(0).getName())
+        .withRecord(OBJECT_MAPPER.createObjectNode()
```

Comment:
can you use 
```
Jsons.jsonNode(
   ImmutableMap.of(
      "date", "2020-03-31T00:00:00Z",
      "HKD", 10,
      "NZD", 700
    )
)
```
Instead?

Reply:
(not sure if that will work since the types are different) but I would like to avoid leaking low level jackson objects.